src/views/List.jsx

```diff
@@ -1,20 +1,43 @@
 import { ListItem } from '../components';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { AddItems } from '../components/AddItems';
 import TextInputElement from '../components/TextInputElement';
+import { comparePurchaseUrgency, urgencyObject } from '../api';
 
 export function List({ data, listPath }) {
 	const [searchItem, setSearchItem] = useState('');
+	const [urgency, setUrgency] = useState(urgencyObject);
+	const listName = listPath.slice(listPath.indexOf('/') + 1);
+
+	useEffect(() => {
+		setUrgency(urgencyObject);
+	}, [urgencyObject]);
+
+	const getClassName = (name) => {
+		const statusArray = Object.entries(urgency).find(([_, items]) => {
+			return Array.from(items).some((item) => item.name === name);
+		});
+		if (!statusArray) {
+			throw new Error(`Failed to get class name of ${name}`);
+		}
+		return statusArray[0];
+	};
 
 	const handleTextChange = (event) => {
 		setSearchItem(event.target.value);
 	};
+	console.log('----');
+	console.log(`before filter: ${data.map((d) => d.name)}`);
+	const filteredItems = data
+		.filter((item) =>
+			item.name.toLowerCase().includes(searchItem.toLowerCase()),
+		)
+		.sort(comparePurchaseUrgency);
 
-	const filteredItems = data.filter((item) =>
-		item.name.toLowerCase().includes(searchItem.toLowerCase()),
-	);
-
-	const listName = listPath.slice(listPath.indexOf('/') + 1);
+	console.log(`filter: ${filteredItems.map((d) => d.name)}`);
+	console.log('----');
+	console.log(`Urgency state object:`);
+	console.log(urgency);
```

Comment:
If we need to keep these console logs in production, can we put them behind a feature flag controlled by a query parameter in the URL? Perhaps something like `?debug=on`.

Reply:
The console logs are just to make testing more visual, I will remove before merging!